packages/compose/src/hooks/use-async-list/index.ts

```diff
@@ -61,10 +61,12 @@ function useAsyncList< T >(
 			if ( list.length <= nextIndex ) {
 				return;
 			}
-			setCurrent( ( state ) => [
-				...state,
-				...list.slice( nextIndex, nextIndex + step ),
-			] );
+			flushSync( () => {
+				setCurrent( ( state ) => [
+					...state,
+					...list.slice( nextIndex, nextIndex + step ),
+				] );
+			} );
```

Comment:
While this PR restores the previous behavior (prior to the upgrade). I think it's still not perfect. Personally, I think we should be calling `asyncQueue.add( {}, append( nextIndex + step ) );` outside the callback. kind of like I did here

https://github.com/WordPress/gutenberg/pull/48085#discussion_r1108263161